adapter/internal/discovery/xds/server.go

```diff
@@ -502,8 +502,15 @@ func DeleteAPIs(vhost, apiName, version string, environments []string, organizat
 				return err
 			}
 			deletedVhosts[vh] = void
+			
+			for val := range deletedVhosts {
+				existingLabels := orgIDOpenAPIEnvoyMap[organizationID][apiIdentifier]
+				if val == vh && len(existingLabels) == 0 {
+					logger.LoggerXds.Infof("Vhost : %v  deleted since there is no gateways assigned to it.", vh)
+					delete(apiToVhostsMap[apiNameVersionHashedID],val)
+				}
+			}
```

Comment:
Can't we get this out from the parent for loop? And avoid checking `val == vh` in if condition?
Just a suggestion

Reply:
we need to check `val == vh` equality to determine deleting vhost properly. Also we need to do this inside the parent loop in order to delete the vhost at the time of gateway deletion (when there are no gateways allocated to the vhost vhost will also get deleted).